header.go

```diff
@@ -101,19 +101,20 @@ func readHeader(r *bufio.Reader, p *Part) (textproto.MIMEHeader, error) {
 		firstColon := bytes.IndexByte(s, ':')
 		firstSpace := bytes.IndexAny(s, " \t\n\r")
 		if firstSpace == 0 {
-			// If there is a single colon followed immediately by a space and there
-			// is no equal sign, or comes before any equal sign, then it should not
-			// be considered as a continuation.
-			colonSpaceIdx := bytes.Index(s, []byte{':', ' '})
-			equalIdx := bytes.IndexByte(s, '=')
-			if bytes.Count(s, []byte{':'}) == 1 && colonSpaceIdx != -1 &&
-				(equalIdx == -1 || equalIdx > colonSpaceIdx) {
-				s = textproto.TrimBytes(s)
-				firstColon = bytes.IndexByte(s, ':')
+			// Do not consider as a continuation if the first whitespace
+			//  in the trimmed line comes right after a colon which,
+			//  in turn, follows at least one character.
+			sTrimmed := textproto.TrimBytes(s)
+			firstTrimmedColon := bytes.IndexByte(sTrimmed, ':')
+			firstTrimmedSpace := bytes.IndexAny(sTrimmed, " \t\n\r")
+			if firstTrimmedSpace > 0 && firstTrimmedColon > 0 &&
```

Comment:
`firstTrimmedSpace > 0` -- I think you can eliminate this check, as the == +1 below implies it.